src/pages/Products.js

```diff
@@ -1,7 +1,14 @@
-import React from 'react'
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { ProductList } from "../components/ProductList";
+import { getProducts } from "../services/products";
+import { fetchProducts, selectProducts } from "../slices/productsSlice";
 
 export const Products = () => {
-  return (
-    <div>Products</div>
-  )
-}
+  const products = useSelector(selectProducts);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    getProducts().then((data) => dispatch(fetchProducts(data)));
+  });
```

Comment:
`  useEffect(() => {
    getProducts().then((data) => dispatch(fetchProducts(data)));
  });`

a useEffect is useful for calling a side effect after a render, this hook has two parameters, the function and an array of dependencies.


- If you don't pass the second parameter then this hook will be called on every render and can affect performance
- if you pass an empty array then it will only be called once when the component is mounted, 
- and if you pass an array with dependencies then it will be called when mounted the component and when any of the variables change from the dependency array.

If we are thinking about performance then it would be good to pass an empty array to be called after the component is mounted